lib/routes/home_route/search_route.dart

```diff
@@ -855,10 +896,21 @@ class _ContentChip extends StatefulWidget {
     Key? key,
     required this.delegate,
     required this.contentType,
-  }) : super(key: key);
+  }) : favoritesChip = false,
+       super(key: key);
+
+  const _ContentChip.favorites({
```

Comment:
There would be a lot of code duplication, or we would have to create a separate abstract widget with an abstract state to inherit from. Both options seem worse than just adding a variable here.

Reply:
There is no need for code duplication or a separate abstract widget, we just have to make the _ContentChip a bit more extensible. I think this improves the code by separating concerns.

<details>
<summary>Here is what I have in mind.</summary>

```dart
class _ContentChip extends StatefulWidget {
  const _ContentChip({
    Key? key,
    required this.delegate,
    required this.contentType,
  }) : super(key: key);

  final _SearchStateDelegate delegate;
  final Type? contentType;

  static const double height = 34.0;

  @override
  _ContentChipState createState() => _ContentChipState();
}

class _ContentChipState extends State<_ContentChip> with SingleTickerProviderStateMixin {

  late AnimationController controller;

  _SearchStateDelegate get delegate => widget.delegate;

  bool get active => delegate.contentType == widget.contentType;

  @override
  void initState() { 
    super.initState();
    controller = AnimationController(vsync: this, duration: const Duration(milliseconds: 300));
    _init();
    if (active) {
      controller.forward();
    }
  }
  
  void _init() {
    delegate.registerChipContext(context, widget.contentType!);
  }

  @override
  void dispose() {
    controller.dispose();
    super.dispose();
  }

  void _handleTap() {
    if (active) {
      delegate.contentType = null;
    } else {
      if (delegate.singleListScrollController.hasClients) {
        delegate.singleListScrollController.jumpTo(0);
      }
      delegate.contentType = widget.contentType;
    }
  }
  
  @override
  Widget build(BuildContext context) {
    if (active) {
      controller.forward();
    } else {
      controller.reverse();
    }
    final colorScheme =  ThemeControl.instance.theme.colorScheme;
    final colorTween = ColorTween(
      begin: colorScheme.secondary,
      end: Constants.Theme.contrast.auto,
    );
    final baseAnimation = CurvedAnimation(
      parent: controller,
      curve: Curves.easeOut,
      reverseCurve: Curves.easeIn,
    );
    final colorAnimation = colorTween.animate(baseAnimation);
    final textColorAnimation = ColorTween(
      begin: colorScheme.onBackground,
      end: textColor,
    ).animate(baseAnimation);
    final splashColorAnimation = ColorTween(
      begin: Constants.Theme.glowSplashColor.auto,
      end: Constants.Theme.glowSplashColorOnContrast.auto,
    ).animate(baseAnimation);
    return SizedBox(
      height: _ContentChip.height,
      child: AnimatedBuilder(
        animation: controller,
        builder: (context, child) => Material(
          color: colorAnimation.value,
          borderRadius: borderRadius,
          shape: border,
          child: NFInkWell(
            borderRadius: borderRadius,
            splashColor: splashColorAnimation.value,
            onTap: _handleTap,
              child: _buildContent(textColorAnimation.value),
            ),
          ),
      ),
    );
  }
  
  ShapeBorder? get border => null;

  BorderRadius? get borderRadius => const BorderRadius.all(Radius.circular(50.0));
  
  Color get textColor => Constants.Theme.contrast.autoReverse;
  
  Widget _buildContent(Color? color) {
    final l10n = getl10n(context);
    final count = delegate.results.map.getValue(widget.contentType)!.length;
    return IgnorePointer(
    child: Theme(
      data: ThemeControl.instance.theme.copyWith(canvasColor: Colors.transparent),
      child: RawChip(
        shape: StadiumBorder(
          side: BorderSide(
              color: Constants.Theme.contrast.auto.withOpacity(0.05),
              width: 1.0
          ),
        ),
        backgroundColor: Colors.transparent,
        label: Text(
          l10n.contentsPlural(count, widget.contentType),
          style: TextStyle(
            color: color,
            fontWeight: FontWeight.w800,
          ),
        ),
      ),
    ),
  );
  }
}

class _FavoritesContentChip extends _ContentChip {
  final borderRadius = null;
  _FavoritesContentChip({
    Key? key,
    required _SearchStateDelegate delegate,
  }) : super(key: key, delegate: delegate, contentType: null);

  @override
  _ContentChipState createState() => _FavoritesContentChipState();
}

class _FavoritesContentChipState extends _ContentChipState {
  void _favoriteListener() {
    setState(() {
      // update since [active] has changed
    });
  }

  @override
  void _init() {
    delegate.showOnlyFavoritesNotifier.addListener(_favoriteListener);
  }
  
  @override
  void dispose() {
    delegate.showOnlyFavoritesNotifier.removeListener(_favoriteListener);
    super.dispose();
  }
  
  @override
  bool get active => delegate.showOnlyFavoritesNotifier.value;

  @override
  void _handleTap() {
    delegate.toggleShowOnlyFavorites();
  }

  @override
  Widget _buildContent(Color? color) => SizedBox(
    width: _ContentChip.height,
    child: Icon(
      active ? Icons.favorite_rounded
          : Icons.favorite_outline_rounded,
      size: 20.0,
      color: color,
    ),
  );

  @override
  StadiumBorder? get border => StadiumBorder(
    side: BorderSide(
      color: Constants.Theme.contrast.auto.withOpacity(0.05),
      width: 1.0,
    ),
  );

  @override
  final borderRadius = null;
  
  @override
  Color get textColor => Colors.redAccent;
}
```
</details>